packages/backend/src/index.test.ts

```diff
@@ -1,8 +1,165 @@
 import { PluginEnvironment } from './types';
+import { resolverResult } from './plugins/plugins_helper/googleAuthResolver';
+import {
+  AuthResolverCatalogUserQuery,
+  AuthResolverContext,
+  BackstageSignInResult,
+  ProfileInfo,
+  SignInInfo,
+} from '@backstage/plugin-auth-node';
+import { TokenParams } from '@backstage/plugin-auth-backend';
+import { Entity } from '@backstage/catalog-model';
 
 describe('test', () => {
   it('unbreaks the test runner', () => {
     const unbreaker = {} as PluginEnvironment;
     expect(unbreaker).toBeTruthy();
   });
 });
+
+describe('providers.google.create.signIn.resolver logic', () => {
```

Comment:
to fix the repetitive logic; use a `beforeEach` stead
```
 let mockProfile, mockSignInInfo, mockContext;

 beforeEach(() => {
    mockProfile = { displayName: 'John Doe', picture: 'https://example.com/avatar.jpg' };
    mockSignInInfo = { profile: mockProfile, result: {} };
    mockContext = {
      issueToken: jest.fn().mockResolvedValue({ token: 'fake token' }),
      findCatalogUser: jest.fn(),
      signInWithCatalogUser: jest.fn(),
    };
  });
```

Reply:
This way the blocks should run before each one of the tests, with the variables set up fresh for each test case, so you've also less redundancy in the code and a more centralized setup that's easier to manage later down the line.